db/migrate/20150930183738_migrate_content_hosts.rb

```diff
@@ -295,6 +297,37 @@ def unregister_all_but_last_system(systems)
     system
   end
 
+  def handle_hypervisors
+    MigrateContentHosts::System.where(:type => HYPERVISOR_CLASS).find_each do |hypervisor|
+      org = hypervisor.environment.organization
+      name = "virt-who-#{hypervisor.name}-#{org.id}"
+
+      host = ::Host.find_by(:name => name)
+      if host.nil?
+        host = MigrateContentHosts::Host.create!(:name => name, :organization => org.becomes(MigrateContentHosts::Organization), :managed => false,
+                                             :type => "Host::Managed", :location => MigrateContentHosts::Location.default_location)
+      elsif host.organization != org
+        logger.warn("Found host with name #{name} but it's in org #{host.organization.name} instead of #{org.name}.")
+        unregister_system(hypervisor)
+        next
+      end
+      hypervisor.host_id = host.id
+      hypervisor.save!
+      create_subscription_facet(host, hypervisor) unless host.subscription_facet
+      update_hypervisor_consumer(hypervisor)
+    end
+  end
+
+  def update_hypervisor_consumer(hypervisor)
+    org = hypervisor.environment.organization
+    hypervisor.content_view = org.default_content_view.becomes(MigrateContentHosts::ContentView)
+    hypervisor.environment = org.library.becomes(MigrateContentHosts::KTEnvironment)
+    hypervisor.save!
+
+    candlepin_environment_id = org.default_content_view.content_view_environments.first.cp_id
+    ::Katello::Resources::Candlepin::Consumer.update(hypervisor.uuid, 'environment' => {'id' => candlepin_environment_id})
```

Comment:
Previously this migration was only reading data, and the rake task handled updates, this introduces doing updates now in the migration?